File: common/src/main/java/store/tacomall/common/db/entity/seckill/Seckill.java
/***
 * @Author: 码上talk|RC
 * @Date: 2020-07-13 14:45:52
 * @LastEditTime: 2020-11-02 15:28:05
 * @LastEditors: 码上talk|RC
 * @Description: 
 * @FilePath: /tacomall-springboot/tacomall-entity/src/main/java/store/tacomall/entity/seckill/Seckill.java
 * @Just do what I think it is right
 */
package store.tacomall.common.db.entity.seckill;

import java.util.Date;

import lombok.Data;
import com.baomidou.mybatisplus.annotation.*;

@Data
public class Seckill {
    @TableId(value = "id", type = IdType.AUTO)
    private int id;

    private int applyTotalCount;

    private int applyUsedCount;

    private Date startTime;

    private Date endTime;

    private int isDelete;

    private Date createTime;

    private Date updateTime;

    private Date deleteTime;

}

File: service/seckill/src/main/resources/lua/seckill.lua
--[[
Author: 码上talk|RC
Date: 2020-12-08 10:08:10
LastEditTime: 2020-12-08 10:08:11
LastEditors: 码上talk|RC
Description: 
FilePath: /tacomall-springcloud/mq/rabbitmq/src/main/resources/lua/seckill.lua
微信:  13680065830
邮箱:  3189482282@qq.com
oops: Just do what I think it is right
--]]
local function split(str, split_char)
    local sub_str_tab = {}
    while true do
        local pos = string.find(str, split_char)
        if not pos then
            table.insert(sub_str_tab, str)
            break
        end
        local sub_str = string.sub(str, 1, pos - 1)
        table.insert(sub_str_tab, sub_str)
        str = string.sub(str, pos + 1, string.len(str))
    end
    return sub_str_tab
end

if redis.call('SISMEMBER', KEYS[1], KEYS[2]) == 1 then
    return -1
end

local table_value = split(redis.call('GET', KEYS[3]), '/')

if tonumber(table_value[2]) <= 0 then
    return -2
end

redis.call('SET', KEYS[3], table_value[1] .. '/' .. table_value[2] - 1)

return redis.call('SADD', KEYS[1], KEYS[2])
